domain/src/main/java/com/example/domain/usecase/document/DocumentUseCaseImpl.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.domain.usecase.document
+
+import com.example.core.lib.utils.IODispatcher
+import com.example.core.models.pagination.PaginationRequest
+import com.example.core.models.response.mapper.mapAndConverterToStateData
+import com.example.core.models.stateData.Either
+import com.example.core.models.stateData.ExceptionState
+import com.example.data.dataSource.booking.BookingDataSource
+import com.example.data.dataSource.tutor.TutorDataSource
+import com.example.data.dataSource.user.UserDataSource
+import com.example.domain.entity.BookingInfoEntity
+import com.example.domain.entity.TutorFavorites
+import com.example.domain.mapper.toDomain
+import com.example.domain.mapper.toFavoriteDomain
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+class DocumentUseCaseImpl @Inject constructor(
+    @IODispatcher private val ioDispatcher: CoroutineContext,
+    private val bookingDataSource: BookingDataSource,
+    private val userDataSource: UserDataSource,
+    private val tutorDataSource: TutorDataSource,
+) : DocumentUseCase {
+    override fun fetchUpComing(time: Long): Flow<Either<ExceptionState, BookingInfoEntity>> =
+        flow {
+            val response = bookingDataSource.fetchUpComing(time)
+            val dataConvert = response.mapAndConverterToStateData {
+                it.toDomain()
+            }
+            emit(dataConvert)
+        }
+
+    override fun fetchTotalTime(): Flow<Either<ExceptionState, Int>> =
+        flow {
+            val response = userDataSource.fetchTotalTime()
+            val dataConvert = response.mapAndConverterToStateData {
+                it.toDomain()
+            }
+            emit(dataConvert)
+        }
+
+    override fun fetchTutors(paginationRequest: PaginationRequest): Flow<Either<ExceptionState, TutorFavorites>> =
+        flow {
+            val response = tutorDataSource.fetchTutors(paginationRequest = paginationRequest)
+            val dataConvert = response.mapAndConverterToStateData {
+                it.toFavoriteDomain()
+            }
+            emit(dataConvert)
+        }
+
+    override fun addFavoriteTutor(tutorId: String): Flow<Either<ExceptionState, Boolean>> = flow {
+        val response = tutorDataSource.addFavoriteTutor(tutorId)
+
+        val dataConvert = response.mapAndConverterToStateData {
+            it.result != 1.0
```

Comment:
Magic number ?

Reply:
hmm, i test the API and I see that:
when I add a tutor to favorite, if the tutor is already existed the response will like this 
```
{
    "message": "Manage success",
    "result": 1
}
```
if add success, the response will like this
```
{
    "message": "Manage success",
    "result": {
        "id": "d5d5efb6-2258-4f38-9406-a06e02a86eef",
        "firstId": "cb9e7deb-3382-48db-b07c-90acf52f541c",
        "secondId": "6ca5c092-76ea-4e72-9c6e-05e2239aa33b",
        "createdAt": "2024-12-11T02:38:49.234Z",
        "updatedAt": "2024-12-11T02:38:49.234Z",
        "secondInfo": null
    }
}
```

Reply:
Can you move to constant?
Example: addFavSuccessCode = 1